test/Directory.Build.props

```
@@ -52,8 +52,23 @@
     <PackageReference Include="Xunit.SkippableFact" Version="1.4.13" />
     <PackageReference Include="GitHubActionsTestLogger" Version="2.3.2" />
     <PackageReference Include="TestableIO.System.IO.Abstractions.TestingHelpers" Version="19.2.29" />
+    <PackageReference Include="coverlet.msbuild" Version="6.0.0">
+      <PrivateAssets>all</PrivateAssets>
+      <IncludeAssets>runtime; build; native; contentfiles; analyzers; buildtransitive</IncludeAssets>
+    </PackageReference>
   </ItemGroup>
 
+
+  <!-- Configuration for Coverlet code coverage -->
+  <PropertyGroup Condition="'$(CollectCoverage)' == 'true'">
+    <DeterministicReport>true</DeterministicReport>
+    <Exclude>[Sentry.Protocol.Test*]*,[xunit.*]*,[System.*]*,[Microsoft.*]*,[Sentry.Test*]*</Exclude>
+    <CoverletOutput>./coverage/</CoverletOutput>
+    <CoverletOutputFormat>opencover</CoverletOutputFormat>
+    <!-- Deterministic report not supported by openCover reporter -->
+    <DeterministicReport>false</DeterministicReport>
```

Cobertura is XML (which codecov expects) and also supports determinstic tests, so lets try: https://github.com/coverlet-coverage/coverlet/blob/cbe464472939490fb8a3add80f44726909886431/Documentation/DeterministicBuild.md#deterministic-report

```suggestion
    <CoverletOutputFormat>cobertura</CoverletOutputFormat>
```